src/ast/mod.rs

```diff
@@ -205,26 +205,17 @@ pub trait Ast<'arena>: std::fmt::Debug + Clone + Eq + Default + std::hash::Hash
 
     /* AST EDITING FUNCTIONS */
 
-    /// Generate an iterator over the possible shorthand [`char`]s that a user could type to replace
-    /// this node with something else.
-    fn replace_chars(&self) -> Box<dyn Iterator<Item = char>>;
-
-    /// Returns whether or not a given [`char`] is in [`Self::replace_chars`]
-    fn is_replace_char(&self, c: char) -> bool {
-        self.replace_chars().any(|x| x == c)
-    }
-
     /// Generate a new node from a [`char`] that a user typed.  If `c` is an element of
-    /// [`get_replace_chars`](Ast::replace_chars), this must return [`Some`] node,
-    /// if it isn't, then this should return [`None`].
+    /// [`valid_chars`](Ast::valid_chars), at current cursor position, [is_valid_child](Ast::is_valid_child) or [is_valid_root](Ast::is_valid_root) they must return [`true`],
+    /// if it isn't, then this should return [`false`].
     fn from_char(&self, c: char) -> Option<Self>;
```

Comment:
I'm a bit confused by this comment... I'm not quite sure what it's trying to say.